api4/hosted_customer.go

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package api4
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/mattermost/mattermost-server/v6/model"
+)
+
+// APIs for self-hosted workspaces to communicate with the backing customer & payments system.
+// Endpoints for cloud installations should not go in this file.
+func (api *API) InitHostedCustomer() {
```

Comment:
Are we planning to add additional purchase endpoints for the self-hosted transactions? Just in general thinking about the naming of additional endpoints here of `/hosted-customer/subscribe`?